File: lab3/app/src/main/java/com/example/myapplication/MainActivity.java
package com.example.myapplication;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;
import android.widget.TextView;

import androidx.activity.EdgeToEdge;
import androidx.activity.result.ActivityResultLauncher;
import androidx.activity.result.contract.ActivityResultContracts;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {

    private TextView resultTextView;
    private ActivityResultLauncher<Intent> launcher;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);

        resultTextView = findViewById(R.id.textViewResult);
        Button button1 = findViewById(R.id.button1);

        launcher = registerForActivityResult(
                new ActivityResultContracts.StartActivityForResult(),
                result -> {
                    if (result.getResultCode() == RESULT_OK && result.getData() != null) {
                        String data = result.getData().getStringExtra(MainActivity2.EXTRA_RESULT);
                        if (data != null && !data.isEmpty()) {
                            resultTextView.setText(data);
                        }
                    }
                }
        );

        button1.setOnClickListener(v -> {
            Intent intent = new Intent(MainActivity.this, MainActivity2.class);
            launcher.launch(intent);
        });

        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });
    }
}

File: lab2/app/src/main/java/com/example/lab2/MainActivity.kt
package com.example.lab2

import android.graphics.Color
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import androidx.activity.ComponentActivity

class MainActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val txtResults = findViewById<TextView>(R.id.txt_results)
        val buttonOk = findViewById<Button>(R.id.bnt_ok)
        val editPassword = findViewById<EditText>(R.id.edit_Password)

        buttonOk.setOnClickListener {
            val pwd = editPassword.text.toString().trim()
            if (pwd == "abc") {
                txtResults.text = "llm是我erzi"
                txtResults.setTextColor(Color.BLACK)
                txtResults.textSize = 20f
            } else {
                txtResults.text = "密码：abc"
                txtResults.setTextColor(Color.rgb(55, 0, 150))
                txtResults.textSize = 30f
            }
        }
    }
}
